src/store/features/blocks/blocks.ts

```diff
@@ -0,0 +1,97 @@
+import tbConf, {ToolboxConfiguration} from "../../../toolboxconfig"
+import {produce} from 'immer'
+import {MutableRefObject} from "react";
+
+// Define actions
+const TOGGLE_CATEGORY = 'category/toggle'
+const TOGGLE_BLOCK_DESCRIPTION = 'block/toggleDescription'
+
+// Type actions
+type ToggleCategory = {
+  type: typeof TOGGLE_CATEGORY,
+  payload: {
+    category: number,
+    blocksDivRef: MutableRefObject<any>
+  }
+}
+
+type ToggleBlockDescription = {
+  type: typeof TOGGLE_BLOCK_DESCRIPTION,
+  payload: {
+    block: number,
+    blockRef: MutableRefObject<any>
+  }
+}
+
+// Action creators
+export const toggleCategoryAction = (categoryId: number, categoryRef: MutableRefObject<any>): ToggleCategory => ({
+  type: TOGGLE_CATEGORY,
+  payload: {
+    category: categoryId,
+    blocksDivRef: categoryRef
+  }
+})
+
+export const toggleBlockDescriptionAction = (blockId: number, blockRef: MutableRefObject<any>): ToggleBlockDescription => ({
+  type: TOGGLE_BLOCK_DESCRIPTION,
+  payload: {
+    block: blockId,
+    blockRef: blockRef
+  }
+})
+
+// Reducers
+type Actions = ToggleCategory | ToggleBlockDescription
+
+let calcHeight = true
+const blocksReducer = (state: ToolboxConfiguration = tbConf, action: Actions) => {
+  switch (action.type) {
+    case 'category/toggle':
```

Comment:
Reuse the constants you've defined:

    case ActionTypes.BlocksCategoryToggle: